WordPress/gutenberg
# [Create Block] Introduce the transformer property.

## What?
This PR introduces the concept of a transformer property to create-block and closes https://github.com/WordPress/gutenberg/issues/55421

## Why?
This provide almost complete control over how a custom template can create plugins/blocks and allows the tool to focus on scaffolding rather than providing a huge amount of options. This offloads customization to the template.

## How?
Creating a default `transformer` property that returns the data sent to it that can be overridden in a custom template.

## Testing Instructions

1. Navigation to `packages/create-block`
2. Add the following code to  the `es5` template in `lib/templates.js` after the `slug` property
3. Run `node ./index transformer-test  --template es5`
4. Confirm that a new plugin is created called `transformer-test-random-string` and anywhere the slug is used in the files has  `-random-string` added to it.

Code to add:
```js
transformer: ( view ) => {
    return {
         ...view,
         slug: `${ view.slug }-random-string`,
    };
 },
